shared/reducers/chat.js

```diff
@@ -151,7 +151,7 @@ function reducer(state: Constants.State = initialState, action: Constants.Action
       const newInbox = action.payload.inbox.map(newRow => {
         const id = newRow.get('conversationIDKey')
         const existingRow = existingRows.find(existingRow => existingRow.get('conversationIDKey') === id)
-        return existingRow ? (existingRow.teamType !== newRow.teamType ? newRow : existingRow) : newRow
+        return existingRow ? (existingRow.version < newRow.version ? newRow : existingRow) : newRow
```

Comment:
@chrisnojima I know you are working on something grand, but this is a temporary fix that I think gets us into a much better state. Now instead of the _ad hoc_ team type check, we check the version, which should be more correct.